lib/node-token-validation/src/network/HttpClient.ts

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { INetworkModule, NetworkRequestOptions, NetworkResponse } from "@azure/msal-common";
+import axios, { AxiosRequestConfig } from "axios";
+import { HttpMethod } from "../utils/Constants";
+
+/**
+ * HttpClient class implements API for network requests
+ */
+export class HttpClient implements INetworkModule {
+
+    /**
+     * Http Get request
+     * @param url 
+     * @param options 
+     * @returns 
+     */
+    async sendGetRequestAsync<T>(url: string, options?: NetworkRequestOptions): Promise<NetworkResponse<T>> {
+        const request: AxiosRequestConfig = {
+            method: HttpMethod.GET,
+            url: url,
+            headers: options && options.headers,
+            validateStatus: () => true
+        };
+
+        const response = await axios(request);
+
+        return {
+            headers: response.headers,
+            body: response.data as T,
+            status: response.status
+        };
+    }
+
+    /**
+     * Http Post request
+     * @param url 
+     * @param options 
+     * @returns 
+     */
+    async sendPostRequestAsync<T>(url: string, options?: NetworkRequestOptions): Promise<NetworkResponse<T>> {
+        const request: AxiosRequestConfig = {
+            method: HttpMethod.POST,
+            url: url,
+            data: (options && options.body) || "",
+            headers: options && options.headers,
+            validateStatus: () => true
+        };
+
+        const response = await axios(request);
+
+        return {
+            headers: response.headers,
+            body: response.data as T,
+            status: response.status
+        };
+    }
+}
```

Comment:
Can we consider `node-fetch`? Also `axios` default implementation does not support proxy out of the box, please check #4447 for details.

Reply:
Offline discussion determined that we would keep `fetch` from node in consideration, but would not immediately upgrade. Support for proxies will come in another PR.